src/jdk.jartool/share/classes/sun/security/tools/jarsigner/Main.java

```diff
@@ -246,7 +249,20 @@ public void run(String args[]) {
             }
 
             if (providerClasses != null) {
-                ClassLoader cl = ClassLoader.getSystemClassLoader();
+                ClassLoader cl;
+                if (pathlist != null) {
+                    String path = null;
+                    path = PathList.appendPath(
+                            path, System.getProperty("java.class.path"));
```

Comment:
Is it more straightforward to use "String path = System.getProperty(...)", by combining line 254 and 255?

Reply:
Seems so. I'll fix it.